hathor/consensus.py

```diff
@@ -242,14 +243,46 @@ def update_voided_info(self, block: Block) -> None:
                     meta = block.get_metadata()
                     if not meta.voided_by:
                         storage._best_block_tips = [block.hash]
+                        self.log.debug('index new winner block', height=meta.height, block=block.hash_hex)
+                        # We update the height cache index with the new winner chain
+                        self.update_block_height_cache_new_chain(meta.height, block)
                 else:
                     storage._best_block_tips = [blk.hash for blk in heads]
+                    # XXX Is it safe to select one of the heads?
+                    best_block = heads[0]
+                    assert best_block.hash is not None
+                    best_meta = best_block.get_metadata()
+                    self.log.debug('index previous best block', height=best_meta.height, block=best_block.hash_hex)
+                    storage.add_to_block_height_index(best_meta.height, best_block.hash)
 
         # Uncomment the following lines to check that the cache update is working properly.
         # You shouldn't run this test in production because it dampens performance.
         #     v = storage.get_best_block_tips(skip_cache=True)
         #     assert v == storage._best_block_tips
 
+    def update_block_height_cache_new_chain(self, height: int, block: Block) -> None:
```

Comment:
I feel this method should be somewhere else. It is clearly related to the index (and not to the consensus).

Reply:
I agree, I'm moving it to `TransactionStorage`.